nixos/modules/virtualisation/vmware-guest.nix

```diff
@@ -38,7 +38,7 @@ in
       };
 
     # Mount the vmblock for drag-and-drop and copy-and-paste.
-    systemd.mounts = [
+    systemd.mounts = mkIf (!cfg.headless) [
```

Comment:
Why does this depend on headless or not? Can't the VM be run headless, but still use this shared folder?

Reply:
When I opened this PR Iooked on the web and that plus the comment above the mount lead me to believe that `vmware-vmblock-fuse` is required specifically for drag & drop. I did not find a lot of info indicating that it was required for shared folders as well.

The one thing that I found now that you're asking that points to `vmhgfs-fuse` needing `vmware-vmblock-fuse` to run as a service for shared folders to work is this section on the arch wiki:
https://wiki.archlinux.org/title/VMware/Install_Arch_Linux_as_a_guest#Systemd_2

The vmware docs [here](https://kb.vmware.com/s/article/60262) and [this](https://github.com/vmware/open-vm-tools/issues/321#issuecomment-460399660) make me think `vmhgfs-fuse` is what's actually required and `vmware-vmblock-fuse` is unrelated.

I did not do any testing with shared folders so far, since I did not think they would be affected.

Reply:
Ah, seems like I misinterpreted what you wrote in the PR description: "Since the vmblock fuse mount and vmware-user-suid-wrapper seem to only be used for shared folders and drag and drop". I'll go ahead and merge this then.

Reply:
Oh, I did not realize that my commit message was inaccurate.

Thanks!